Backend/app/rider-platform/rider-app/Main/src/API/UI/Registration.hs

```diff
@@ -63,9 +81,9 @@ handler =
     :<|> resend
     :<|> logout
 
-auth :: DRegistration.AuthReq -> Maybe Version -> Maybe Version -> FlowHandler DRegistration.AuthRes
-auth req mbBundleVersion =
-  withFlowHandlerAPI . DRegistration.auth req mbBundleVersion
+auth :: DRegistration.AuthRawReq -> Maybe Text -> Maybe Version -> Maybe Version -> FlowHandler DRegistration.AuthRes
```

Comment:
The reason I used the same API is because frontend then needs to still call the same API with the same payload and an optional **x-sdk-authorization** header. Based on if the header is passed in the request body and the signature is verified with the payload, we can return the authType as DIRECT to frontend and skip the verify OTP step.
Else, we can return authType as OTP and frontend has to call verify OTP api.